### ISSUE TITLE
Warn users before executing an automatic change of context

### ISSUE THREAD

**Issue by hostle83** [DSpace/dspace-angular#3094]
**Describe the bug**
According to WCAG 2 Success Criterion: 3.2.2 On Input (A), users should be warned before a change of context occurs on a site. As an example, when searching (https://sandbox.dspace.org/search?query=), use of the slider functionality in the Date filter reloads the page without giving the user a warning of that change or giving them the option to activate the reload themselves. See https://www.w3.org/WAI/WCAG21/Understanding/on-input.html for reference. 

![image](https://github.com/DSpace/dspace-angular/assets/1933584/d92cf11c-a8a7-4721-9ec8-19b4e3d46eea)

This behavior is consistent across all search filters. 

**To Reproduce**
1. Visit the search page on the sandbox, demo, or localhost (.../search?query=)
2. Expand any search filter
3. Adjust the search by using any filter, such as using the date slider to filter search results --> page reloads each time slider is used; the same holds true for all other search filters

**Expected behavior**
Either warn the user of the change of context (reload) beforehand, or give the user to activate the change themselves by providing a button or similar functionality

**Comment by atarix83** [comment]
@tdonohue @hostle83 

4science is interested in working on this task.
In order to fix the problem we'd add an additional button to apply the date range filter. Estimation is about 6h

**Comment by tdonohue** [comment]
Closing as fixed by #3236.  Ported to 8.x and 7.6.x.
